src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,231 @@
+      val persistFirst = redownloadService.persistOwnSnapshot(2L, firstSnapshot)
+      val persistSecond = redownloadService.persistOwnSnapshot(2L, secondSnapshot)
       val check = redownloadService.ownSnapshots.get.map(_.get(2L))
 
-      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe "aaaa".some
+      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe RecentSnapshot("aaaa", 2L, Map.empty).some
     }
   }
 
   "getOwnSnapshots" - {
     "should return empty map if there are no own snapshots" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
 
       val check = redownloadService.getOwnSnapshots()
 
       check.unsafeRunSync shouldBe Map.empty
     }
 
     "should return all own snapshots if they exist" in {
-      val redownloadService = RedownloadService[IO]()
-
-      val persistFirst = redownloadService.persistOwnSnapshot(2L, "aaaa")
-      val persistSecond = redownloadService.persistOwnSnapshot(4L, "bbbb")
+      val redownloadService = RedownloadService[IO](dao)
+      val firstSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
+      val secondSnapshot = RecentSnapshot("bbbb", 4L, Map.empty)
+      val persistFirst = redownloadService.persistOwnSnapshot(2L, firstSnapshot)
+      val persistSecond = redownloadService.persistOwnSnapshot(4L, secondSnapshot)
       val check = redownloadService.getOwnSnapshots()
 
-      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> "aaaa", 4L -> "bbbb")
+      (persistFirst >> persistSecond >> check).unsafeRunSync shouldBe Map(2L -> RecentSnapshot("aaaa", 2L, Map.empty),
+                                                                          4L -> RecentSnapshot("bbbb", 4L, Map.empty))
     }
   }
 
   "getOwnSnapshot" - {
     "should return hash if snapshot at given height exists" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
 
-      val persist = redownloadService.persistOwnSnapshot(2L, "aaaa")
+      val persist = redownloadService.persistOwnSnapshot(2L, newSnapshot)
       val check = redownloadService.getOwnSnapshot(2L)
 
-      (persist >> check).unsafeRunSync shouldBe "aaaa".some
+      (persist >> check).unsafeRunSync shouldBe RecentSnapshot("aaaa", 2L, Map.empty).some
     }
 
     "should return None if snapshot at given height does not exist" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](dao)
 
       val check = redownloadService.getOwnSnapshot(2L)
 
       check.unsafeRunSync shouldBe none[String]
     }
   }
 
+  "fetchPeersProposals" - {
+    "should update peersProposals" in {
+      val redownloadService = RedownloadService[IO](dao)
+      val updateNewProps = redownloadService.updatePeerProps(proposals)
```

Comment:
Why are you triggering `updatePeerProps` if you're testing `fetchPeersProposals`? You shouldn't call internal methods but only the `fetchPeersProposals` and check for the proper outcome and behavior. Consider sth like this, which I wrote for previous implementation:

```
  "should fetch own proposals of all the peers" in {
      val peerInfo = Map(Id("node1") -> mock[PeerData], Id("node2") -> mock[PeerData])
      peerInfo.values.foreach { peer =>
        peer.client shouldReturn mock[APIClient]
        peer.client.getNonBlockingF[IO, Map[Long, String]](*, *, *)(*)(*, *, *) shouldReturnF Map.empty
      }

      cluster.getPeerInfo shouldReturnF peerInfo

      redownloadService.fetchPeersProposals().unsafeRunSync

      peerInfo.values.foreach { peer =>
        peer.client.getNonBlockingF[IO, Map[Long, String]]("/snapshot/own", *, *)(*)(*, *, *).was(called)
      }
    }

  "should modify the local proposals of peers" in {
      val peer1 = mock[PeerData]
      val peer2 = mock[PeerData]

      val peerInfo = Map(Id("node1") -> peer1, Id("node2") -> peer2)
      val proposals = Map(2L -> "aa", 4L -> "bb")

      val ids = List(Id("node1"), Id("node2"))

      peerInfo.values.zip(ids).foreach {
        case (peer, id) =>
          peer.client shouldReturn mock[APIClient]
          peer.client.id shouldReturn id
          peer.client.getNonBlockingF[IO, Map[Long, String]](*, *, *)(*)(*, *, *) shouldReturnF proposals
      }

      cluster.getPeerInfo shouldReturnF peerInfo

      val fetch = redownloadService.fetchPeersProposals()
      val check = redownloadService.peersProposals.get

      (fetch >> check).unsafeRunSync shouldBe Map(
        Id("node1") -> proposals,
        Id("node2") -> proposals
      )
    }

  "should not fail if at least one peer did not respond" in {
      val peer1 = mock[PeerData]
      val peer2 = mock[PeerData]

      val peerInfo = Map(Id("node1") -> peer1, Id("node2") -> peer2)
      val proposals = Map(2L -> "aa", 4L -> "bb")

      peer1.client shouldReturn mock[APIClient]
      peer1.client.id shouldReturn Id("node1")
      peer1.client.getNonBlockingF[IO, Map[Long, String]](*, *, *)(*)(*, *, *) shouldReturnF proposals

      peer2.client shouldReturn mock[APIClient]
      peer2.client.id shouldReturn Id("node2")
      peer2.client.getNonBlockingF[IO, Map[Long, String]](*, *, *)(*)(*, *, *) shouldReturn IO.raiseError(
        new Throwable("error")
      )

      cluster.getPeerInfo shouldReturnF peerInfo

      val fetch = redownloadService.fetchPeersProposals()
      val check = redownloadService.peersProposals.get

      (fetch >> check).unsafeRunSync shouldBe Map(
        Id("node1") -> proposals,
        Id("node2") -> Map.empty
      )
    }
```